ksql-streams/src/test/java/io/confluent/ksql/execution/streams/StreamSinkBuilderTest.java

```diff
@@ -0,0 +1,192 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import io.confluent.ksql.execution.context.QueryContext;
+import io.confluent.ksql.execution.plan.DefaultExecutionStepProperties;
+import io.confluent.ksql.execution.plan.ExecutionStep;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.StreamSink;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.serde.Format;
+import io.confluent.ksql.serde.FormatInfo;
+import io.confluent.ksql.serde.KeyFormat;
+import io.confluent.ksql.serde.KeySerde;
+import io.confluent.ksql.serde.SerdeOption;
+import io.confluent.ksql.serde.ValueFormat;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.kstream.ValueMapper;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class StreamSinkBuilderTest {
+  private static final String TOPIC = "TOPIC";
+  private static final LogicalSchema SCHEMA = LogicalSchema.builder()
+      .valueColumn("BLUE", SqlTypes.BIGINT)
+      .valueColumn("GREEN", SqlTypes.STRING)
+      .build()
+      .withMetaAndKeyColsInValue();
+  private static final PhysicalSchema PHYSICAL_SCHEMA =
+      PhysicalSchema.from(SCHEMA.withoutMetaAndKeyColsInValue(), SerdeOption.none());
+  private static final KeyFormat KEY_FORMAT = KeyFormat.nonWindowed(FormatInfo.of(Format.KAFKA));
+  private static final ValueFormat VALUE_FORMAT = ValueFormat.of(FormatInfo.of(Format.JSON));
+
+  @Mock
+  private KsqlQueryBuilder queryBuilder;
+  @Mock
+  private KeySerdeFactory<Struct> keySerdeFactory;
+  @Mock
+  private KStream<Struct, GenericRow> stream;
+  @Mock
+  private ExecutionStep<KStream<Struct, GenericRow>> source;
+  @Mock
+  private KeySerde<Struct> keySerde;
+  @Mock
+  private Serde<GenericRow> valSerde;
+  @Captor
+  private ArgumentCaptor<ValueMapper<GenericRow, GenericRow>> mapperCaptor;
+
+  private final QueryContext queryContext =
+      new QueryContext.Stacker(new QueryId("qid")).push("sink").getQueryContext();
+
+  private StreamSink<Struct> sink;
+
+  @Rule
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
```

Comment:
Use of `MockitoRule` triggers spotbugs errors for me locally.  Can we use `@RunWith(MockitoJUnitRunner.class)` instead please?